src/overmap.cpp

```diff
@@ -4680,6 +4687,36 @@ void overmap::place_lakes()
     }
 }
 
+// helper function for code deduplication, as it is needed multiple times
+float overmap::calculate_ocean_gradient( const point_om_omt &p, const point_abs_om this_om )
```

Comment:
Ultimately actually, I want to move the noise gradient effect into the overmap_noise.cpp function for oceans. If you'd like, you're welcome to do that here.

Reply:
to be honest, I am very wary of introducing sweeping code changes, especially if they require adding variables across namespaces. the ocean class at overmap_noise.cpp does not seem to have a settings variable. I'll keep it as is for now, feel free to move it later.